### DISCUSSION TITLE
Meaning of x tensor inside ODE system

### DISCUSSION THREAD

**Discussion by asmmmpt** [lululxvi/deepxde#945] [Q&A]
Hello Lu.

I am trying to develop a SBINN using the DeepXDE library. In my system I need to integrate one of the variables. For this I extract the values of the time tensor t and variable G_PL as numpys arrays. However I noticed that the length of t is not as expected. It seems that the length of t inside the ODE system is number of initial conditions + the length of the observed time * (number of observed variables +1), with this last 1 appearing from anchors.

I do not understand the meaning of this and would like to know which portion of t and G_PL I can use to integrate. If there is another way to integrate a state variable (from current time - window until current time), that would also work for me.

Thank you in advance.

**Comment by lululxvi** [comment]
> It seems that the length of t inside the ODE system is number of initial conditions + the length of the observed time * (number of observed variables +1), with this last 1 appearing from anchors.

It is correct. It is designed in this way.

> would like to know which portion of t and G_PL I can use to integrate. If there is another way to integrate a state variable (from current time - window until current time), that would also work for me.

There is no guarantee of the order and values of x during training. To integrate, you can check https://deepxde.readthedocs.io/en/latest/demos/pinn_forward.html#integro-differential-equations

**Reply by asmmmpt** [reply]
Thank you for your reply.
If I understand correctly, the link provided would be to a forward problem of finding the function whose integral is given by the defined equations.
In my problem, I am solving an inverse problem of estimating parameters for a biological system. In this system one of the variables, A, is defined with a PID contoller of another variable, B. Therefore I think I need to numerically integrate this variable as I do not have any kind of equations for the integral.

**Reply by lululxvi** [reply]
> If I understand correctly, the link provided would be to a forward problem of finding the function whose integral is given by the defined equations.

No. It includes numerical integration of the variable.
